lib/nerves_hub_web/channels/device_socket.ex

```diff
@@ -15,7 +16,44 @@ defmodule NervesHubWeb.DeviceSocket do
   # Default 90 seconds max age for the signature
   @default_max_hmac_age 90
 
+  defoverridable init: 1, handle_info: 2, terminate: 2
+
+  @impl true
+  def init(state) do
+    {:ok, {other, socket}} = super(state)
+    socket = on_connect(socket)
+    {:ok, {other, socket}}
+  end
```

Comment:
```suggestion
  def init(state_tuple) do
    {:ok, {state, socket}} = super(state_tuple)
    socket = on_connect(socket)
    {:ok, {state, socket}}
  end
```
Naming suggestion to help our future selves

Reply:
Also, am I understanding right that `init` callback is coming _after_ `connect`? If that's the case, maybe we want to order our callback functions here in the perceived order?

Reply:
You are right, and hmmmm. I put `on_connect` after `init` to make sure the socket was initialized correctly first,. and so we can pass the socket (with its assigns) to `on_connect`